snorkel/learning/gen_learning.py

```diff
@@ -367,31 +365,45 @@ def marginals(self, L):
                                 logp_false += self.weights.dep_reinforcing[j, k]
 
                 marginals[i] = 1 / (1 + np.exp(logp_false - logp_true))
+            return marginals
 
         # Categorical setting
         else:
-            marginals = np.zeros((m, self.cardinality), dtype=np.float64)
+            all_marginals = []
+
+            # Get the marginal (posterior) probability for each candidate
             for i in range(m):
+                marginals = np.zeros(self.cardinalities[i], dtype=np.float64)
```

Comment:
Do we need to tell users how to interpret indices of this vector if their LFs output strings, etc.? (Think I already saw some comment about this...)